src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Integer.cs

```diff
@@ -78,5 +78,29 @@ private void WriteUnsignedInteger(CborMajorType type, ulong value)
                 _offset += 8;
             }
         }
+
+        private int GetIntegerEncodingLength(ulong value)
+        {
+            if (value < (byte)CborAdditionalInfo.Additional8BitData)
+            {
+                return 1;
+            }
+            else if (value <= byte.MaxValue)
+            {
+                return 2;
+            }
+            else if (value <= ushort.MaxValue)
+            {
+                return 3;
+            }
+            else if (value <= uint.MaxValue)
+            {
+                return 5;
+            }
+            else
+            {
+                return 9;
```

Comment:
For some reason I did a mild double-take at 9.  These are probably fine as-is, but consider replacing 2/3/5/9 with `sizeof(comparedType) + 1`, which the compiler will fold back into 2/3/5/9.  (9 being implicitly `sizeof(ulong) + 1`)